storage/diemdb/src/lib.rs

```diff
@@ -328,6 +328,33 @@ impl DiemDB {
         update_rocksdb_properties(&self.db)
     }
 
+    /// Use our current latest ledger version for unspecified ledger version params.
+    /// Validate that ledger and proof version params are in serviceable ranges.
+    fn version_params(
+        &self,
+        ledger_version: Option<Version>,
+        proof_version: Option<Version>,
+    ) -> Result<(Version, Version)> {
+        // Observe a snapshot of the DB's latest ledger version
+        let latest_ledger_version = self
+            .ledger_store
+            .get_latest_ledger_info()?
+            .ledger_info()
+            .version();
+
+        let ledger_version = ledger_version.unwrap_or(latest_ledger_version);
```

Comment:
The ledger_version was designed to be decided from the caller side, who will get it from the user request or from a DB call, but the point is, the caller side makes sure all relevant requests uses the same ledger_version, so that the responses of the DB represent a consistent view of the chain. I think the design still makes sense.

Reply:
done